src/rust/engine/process_execution/src/lib.rs

```diff
@@ -470,6 +484,14 @@ impl From<ProcessResultMetadata> for ExecutedActionMetadata {
   }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ProcessResultSource {
+  RanLocally,
+  RanRemotely,
+  HitLocally,
+  HitRemotely,
+}
```

Comment:
I expected to see `memoized` here too?

Reply:
Memoized is... hard to implement. But it also applies to all `Node`s in the graph, rather than just to `Process`es (and these sources don't apply to other `Node`s).

As it stands, no workunits are created at all when a `Node` is memoized (so a clean run might have 10k workunits, while a warm/noop run contains 0 workunits). Maybe we can find a way to do it at some point in the future though (we'll likely be forced to if we want to use these sources in the UI at all).